bitcoin/src/blockdata/script/borrowed.rs

```diff
@@ -133,13 +133,13 @@ impl Script {
     }
 }
 
+/// Creates a new script builder
+pub fn builder() -> Builder { Builder::new() }
```

Comment:
This entirely loses the purpose of the method - to be able to call `Script::builder()`. If we cannot change it then we should remove it (but I really, really, hate that, so I'm willing to take a look at the macro to solve it).

Reply:
I'd appreciate that, the macro is embarrassingly bad.

Reply:
Are we willing to accept changing all `foo(&self)` to `foo(self: &Self)`? If so I've managed to trim the macro down (and it supports `Script::builder` also).
```rust
/// Defines an trait `$trait_name` and implements it for `ty`, used to define extension traits.
macro_rules! define_extension_trait {
    ($(#[$($trait_attrs:tt)*])* $trait_vis:vis trait $trait_name:ident impl for $ty:ident {
        $(
            $(#[$($fn_attrs:tt)*])*
            fn $fn:ident$(<$($gen:ident: $gent:ident),*>)?($($param_name:ident: $param_type:ty),* $(,)?) $( -> $ret:ty )? $body:block
        )*
    }) => {
        $(#[$($trait_attrs)*])* $trait_vis trait $trait_name {
            $(
                $(#[$($fn_attrs)*])*
                fn $fn$(<$($gen: $gent),*>)?($($param_name: $param_type),*) $( -> $ret )?;
            )*
        }

        impl $trait_name for $ty {
            $(
                fn $fn$(<$($gen: $gent),*>)?($($param_name: $param_type),*) $( -> $ret )? $body
            )*
        }
    };
}
pub(crate) use define_extension_trait;
```

Reply:
I raised #3137 instead of discussing here.